python/cppyy/numba_ext.py

```diff
@@ -130,14 +160,57 @@ def get_call_type(self, context, args, kwds):
         except KeyError:
             pass
 
-        ol = CppFunctionNumbaType(self._func.__overload__(*(numba2cpp(x) for x in args)), self._is_method)
+        args_cpp = []
+        for i, arg in enumerate(list(args)):
+            # If the user explicitly passes an argument using numba CPointer, the regex match is used
+            # to detect the pass by reference since the dispatcher always returns typeref[val*]
+            match = re.search(r"typeref\[(.*?)\*\]", str(arg))
+            if match:
+                literal_val = match.group(1)
+                arg_type = numba.typeof(eval(literal_val))
+                args_cpp.append(to_ref(numba2cpp(arg_type)))
+            else:
+                args_cpp.append(numba2cpp(arg))
+        args_combinations = itertools.product(*args_cpp)
+
+        for arg_combo in args_combinations:
+            if len(arg_combo) >= 1:
+                signature = ", ".join(arg_combo)
+            elif len(arg_combo) == 0:
+                signature = ""
+            try:
+                ol = CppFunctionNumbaType(self._func.__overload__(signature), self._is_method)
+                self._arg_set_matched = arg_combo
+                break
+            except:
+                # TODO : Find a better way to handle this error case. Despite a succesful __overload__ match,
+                #  the error could arise from failures in the lowering pass
+                raise RuntimeError("__overload__ call failed")
+                pass
 
         thistype = None
         if self._is_method:
             thistype = nb_types.voidptr
 
+        return_type = ol._func.__cpp_reflex__(cpp_refl.RETURN_TYPE)
+        is_reference_return = False
+
+        # TODO : This works for builtin reference type returns but not for objects
+        #  Use typeof_scope() and eventually merge this logic into cpp2numba
+        if str(type(return_type)) == "<class 'str'>":
+            if return_type.endswith("&"):
```

Comment:
This type of info should come from the `__reflex__` API.

Reply:
Fixed with latest commits, succesfully njits const pass-by-refs, as well as pointers